tests/auto_generated/SimplicialComplex_test_autogen.py

```diff
@@ -0,0 +1,126 @@
+        cpp_code = 'TEST_CASE("{test_func}_{mesh_name}", "[simplicial_complex][{test_func}][2D][auto_generated]")\n'.format(mesh_name=mesh_name.split(".")[0], test_func = test_function_name)+\
+        '{\n'+\
+        '    RowVectors3d V;\n'+\
+        '    RowVectors3l F;\n'+\
+        '    std::string name = "/{mesh_file}";\n'.format(mesh_file = mesh_name)+\
+        '    std::string path;\n'+\
+        '    path.append(WMTK_DATA_DIR);\n'+\
+        '    path.append(name);\n'+\
+        '    igl::read_triangle_mesh(path, V, F);\n'+\
+        '    tests::DEBUG_TriMesh m;\n'+\
+        '    m.initialize(F);\n'+\
+        '\n'+\
+        '    Tuple t;\n'+\
+        '    std::vector<std::vector<long>> sc_v, sc_e, sc_f;\n'
+        file.write(cpp_code)
+
+        # print(v_samples)
+        for v in v_samples:
+            file.write("    t = m.tuple_from_id(PV, {});\n".format(v[0]))
+            file.write("    sc_v = m.get_sorted_simplicial_complex(SimplicialComplex::{test_func}(m, Simplex(PV, t)).get_simplex_vector());\n".format(test_func = test_function_name))
+            sc = test_function(v)
+            sc = sorted(sc, key=lambda x: (len(x),x))
+            file.write("    REQUIRE(sc_v.size() == {});\n".format(len(sc)))
+            for simplex in sc:
+                for i in range(len(simplex)):
+                    file.write("    CHECK(sc_v[{}][{}] == {});\n".format(sc.index(simplex), i, simplex[i]))
+            file.write("\n")
+        
+        # print(e_samples)
+        for e in e_samples:
+            file.write("    t = m.edge_tuple_from_vids({},{});\n".format(e[0], e[1]))
+            file.write("    sc_e = m.get_sorted_simplicial_complex(SimplicialComplex::{test_func}(m, Simplex(PE, t)).get_simplex_vector());\n".format(test_func = test_function_name))
+            sc = test_function(e)
+            sc = sorted(sc, key=lambda x: (len(x),x))
+            file.write("    REQUIRE(sc_e.size() == {});\n".format(len(sc)))
+            for simplex in sc:
+                for i in range(len(simplex)):
+                    file.write("    CHECK(sc_e[{}][{}] == {});\n".format(sc.index(simplex), i, simplex[i]))
+            file.write("\n")
+
+        # print(f_samples)
+        for f in f_samples:
+            file.write("    t = m.face_tuple_from_vids({}, {}, {});\n".format(f[0], f[1], f[2]))
+            file.write("    sc_f = m.get_sorted_simplicial_complex(SimplicialComplex::{test_func}(m, Simplex(PF, t)).get_simplex_vector());\n".format(test_func = test_function_name))
+            sc = test_function(f)
+            sc = sorted(sc, key=lambda x: (len(x),x))
+            file.write("    REQUIRE(sc_f.size() == {});\n".format(len(sc)))
+            for simplex in sc:
+                for i in range(len(simplex)):
+                    file.write("    CHECK(sc_f[{}][{}] == {});\n".format(sc.index(simplex), i, simplex[i]))
+            file.write("\n")
+        
+        file.write("}\n")
+
+
+
+# Parameters
+path = '../../data/'
```

Comment:
this path might not be canonical so it should be an argument for the script with some instructions on what should be found in the path (such as being some subfolder of WMTK_DATA_DIR?)